pandas/core/series.py

```diff
@@ -2227,20 +2245,36 @@ def combine(self, other, func, fill_value=np.nan):
         Series.combine_first : Combine Series values, choosing the calling
             Series's values first
         """
+        self_is_ext = is_extension_array_dtype(self.values)
+        if fill_value is None:
+            if self_is_ext:
```

Comment:
I thikn `na_value_for_dtype` is appropriate here, probably with `compat=False`.

Reply:
I made that change.